### PR TITLE
Add declarative api to invalidate queries in mutations

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

In this PR:

1. We added support to declaratively invalidate queries after making a mutation
2. We pull ReactQueryDevtools up in the component tree to make it available during dashboard loading

Also, this PR removes some rules in eslint

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/ide-desktop/eslint.config.js [hunk 1]
```diff
@@ -87,10 +87,6 @@ const RESTRICTED_SYNTAXES = [
         selector: `:matches(ImportDefaultSpecifier[local.name=/^${NAME}/i], ImportNamespaceSpecifier > Identifier[name=/^${NAME}/i])`,
         message: `Don't prefix modules with \`${NAME}\``,
     },
-    {
-        selector: 'TSTypeLiteral',
-        message: 'No object types - use interfaces instead',
```
Comment (somebody1234) [inline app/ide-desktop/eslint.config.js]: not sure it's worth removing this rule just for a single type

File: app/ide-desktop/eslint.config.js [hunk 2]
```diff
@@ -137,14 +133,6 @@ const RESTRICTED_SYNTAXES = [
         selector: `TSAsExpression:not(:has(TSTypeReference > Identifier[name=const]))`,
         message: 'Avoid `as T`. Consider using a type annotation instead.',
     },
-    {
-        selector: `:matches(\
-            TSUndefinedKeyword,\
-            Identifier[name=undefined],\
-            UnaryExpression[operator=void]:not(:has(CallExpression.argument)), BinaryExpression[operator=/^===?$/]:has(UnaryExpression.left[operator=typeof]):has(Literal.right[value=undefined])\
-        )`,
-        message: 'Use `null` instead of `undefined`, `void 0`, or `typeof x === "undefined"`',
```
Comment (somebody1234) [inline app/ide-desktop/eslint.config.js]: these other two lints aren't related to this PR - if you want to remove them they should at least be related to the PR imo

File: app/ide-desktop/lib/dashboard/src/App.tsx [hunk 3]
```diff
@@ -176,7 +175,7 @@ export default function App(props: AppProps) {
   const routerFuture: Partial<router.FutureConfig> = {
     /* we want to use startTransition to enable concurrent rendering */
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/App.tsx]: comment no longer applies

File: app/ide-desktop/lib/dashboard/src/reactQueryClient.ts [hunk 4]
```diff
@@ -6,13 +6,75 @@
...
+        )
+
+        for (const queryKey of invalidatesToIgnore) {
+          void queryClient.invalidateQueries({
+            predicate: query => reactQuery.matchQuery({ queryKey }, query),
+          })
+        }
+
+        if (invalidatesToAwait.length > 0) {
+          // eslint-disable-next-line no-restricted-syntax
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/reactQueryClient.ts:67]: it's not clear what's the restricted syntax here, if it's causing issues a comment should be added to explain

File: app/ide-desktop/lib/dashboard/src/reactQueryClient.ts [hunk 5]
```diff
@@ -6,13 +6,75 @@
...
+              })
+            )
+          )
+        }
+      },
+    }),
     defaultOptions: {
       queries: {
+        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+        staleTime: 2 * 60 * 1000, // 2 minutes
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/reactQueryClient.ts]: according to convention this should be a constant `DEFAULT_QUERY_STALE_TIME_MS`

**Review by somebody1234 — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/reactQueryClient.ts (hunk 4 above)
Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/reactQueryClient.ts:67]: Early return

File: app/ide-desktop/eslint.config.js (hunk 1 above)
Comment (MrFlashAccount) [reply app/ide-desktop/eslint.config.js]: This pr is a part of larger User Groups PR, I decided to extract eslint changes here to discuss them.

Comment (MrFlashAccount) [reply app/ide-desktop/eslint.config.js]: I actively use inline object types there(e.g. https://github.com/enso-org/enso/pull/10202/files#diff-da2aef9f2124ef55356bc2554513a6034b842304d0a73f7efa8c872f4e252ab7R104), extracting these types into separate interfaces is tedious

File: app/ide-desktop/eslint.config.js (hunk 2 above)
Comment (MrFlashAccount) [reply app/ide-desktop/eslint.config.js]: See: https://github.com/enso-org/enso/pull/10200#discussion_r1630713853

File: app/ide-desktop/eslint.config.js (hunk 1 above)
Comment (somebody1234) [reply app/ide-desktop/eslint.config.js]: they're not supposed to be separate interfaces, this is what `extends` is for in interfaces.

File: app/ide-desktop/eslint.config.js (hunk 2 above)
Comment (somebody1234) [reply app/ide-desktop/eslint.config.js]: it's not productive to discuss changes in a PR where the effects are not visible IMO

Comment (MrFlashAccount) [reply app/ide-desktop/eslint.config.js]: Ok lets omit them

**Pull request merged**